rllib/utils/actor_manager.py

```diff
@@ -400,6 +411,37 @@ def __fetch_result_and_mark_state(
 
         return ready, remote_results
 
+    def _filter_func_and_remote_actor_id_by_state(
+        self,
+        func: Union[Callable[[Any], Any], List[Callable[[Any], Any]]],
+        remote_actor_ids: List[int],
+    ):
+        """Filter out func and remote worker ids by actor state.
+
+        Args:
+            func: A single, or a list of Callables.
+            remote_worker_ids: IDs of potential remote workers to apply func on.
+
+        Returns:
+            A tuple of (filtered func, filtered remote worker ids).
+        """
+        if isinstance(func, list):
+            # We are given a list of functions to apply.
+            # Need to filter the functions together with worker IDs.
+            temp_func = []
+            temp_remote_actor_ids = []
+            for f, i in zip(func, remote_actor_ids):
```

Comment:
you need to make sure that remote_actor_ids and func have the same length. `zip` does not catch this error and will silently pair the two up to the shorter list and you won't error out here. Somewhere else things would break and you have no idea why.

Reply:
sure.